python/hsfs/core/vector_server.py

```diff
@@ -1010,51 +1112,68 @@ def _set_default_client(
-        self, rows: Union[dict, pd.DataFrame], request_parameter: Dict[str, Any]
+        self,
+        rows: Union[dict, pd.DataFrame],
+        request_parameter: Dict[str, Any],
+        transformation_context: Dict[str, Any] = None,
     ) -> dict:
         _logger.debug("Applying On-Demand transformation functions.")
         for tf in self._on_demand_transformation_functions:
-            if (
-                tf.hopsworks_udf.execution_mode.get_current_execution_mode(online=True)
-                == UDFExecutionMode.PANDAS
-            ):
-                # Check if feature provided as request parameter if not get it from retrieved feature vector.
-                features = [
-                    pd.Series(request_parameter[feature])
-                    if feature in request_parameter.keys()
-                    else (
-                        pd.Series(
-                            rows[feature]
-                            if (not isinstance(rows[feature], pd.Series))
-                            else rows[feature]
-                        )
+            # Setting transformation function context variables.
+            tf.hopsworks_udf.transformation_context = transformation_context
+
+            # Check if feature provided as request parameter in prefixed or unprefixed format if not get it from retrieved feature vector.
+            features = []
+            for (
+                unprefixed_feature
+            ) in tf.hopsworks_udf.unprefixed_transformation_features:
+                # Check if the on-demand feature has a prefix. If it does, compute the prefixed feature name.
+                if tf.hopsworks_udf.feature_name_prefix:
+                    prefixed_feature = (
+                        tf.hopsworks_udf.feature_name_prefix + unprefixed_feature
                     )
-                    for feature in tf.hopsworks_udf.transformation_features
-                ]
-            else:
-                # No need to cast to pandas Series for Python UDF's
-                features = [
-                    request_parameter[feature]
-                    if feature in request_parameter.keys()
-                    else rows[feature]
-                    for feature in tf.hopsworks_udf.transformation_features
-                ]
+                else:
+                    prefixed_feature = unprefixed_feature
+
+                # Check if the prefixed feature name is provided as a request parameter, if so then use it. Otherwise if the unprefixed feature name is provided as a request parameter and use it. Else fetch the feature from the retrieved feature vector.
+                if prefixed_feature in request_parameter.keys():
+                    feature_value = request_parameter[prefixed_feature]
+                elif unprefixed_feature in request_parameter.keys():
+                    feature_value = request_parameter[unprefixed_feature]
+                else:
+                    feature_value = rows[prefixed_feature]
```

Comment:
could replace it with:
```suggestion
                feature_value = request_parameter.get(prefixed_feature, 
                 request_parameter.get(unprefixed_feature, 
                 rows.get(prefixed_feature)))
```

Reply:
Replaced as per suggestion.